src/cargo/sources/registry/mod.rs

```diff
@@ -213,17 +212,121 @@ pub struct RegistryConfig {
     pub api: Option<String>,
 }
 
-#[derive(Deserialize)]
-struct RegistryPackage<'a> {
+pub struct RegistryPackage<'a> {
     name: Cow<'a, str>,
     vers: Version,
     deps: DependencyList,
     features: BTreeMap<String, Vec<String>>,
     cksum: String,
     yanked: Option<bool>,
-    #[serde(default)] links: Option<String>,
+    links: Option<String>,
 }
 
+pub struct RegistryPackageDefault(pub SourceId);
+
+impl<'de> de::DeserializeSeed<'de> for RegistryPackageDefault {
+    type Value = RegistryPackage<'de>;
+
+    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        return deserializer.deserialize_map(Visitor(self.0));
```

Comment:
When implementing DeserializeSeed I typically reuse the same type for the DeserializeSeed impl and the Visitor impl, so this would be `deserializer.deserialize_map(self)`.